src/uid2StorageManager.ts

```diff
@@ -0,0 +1,47 @@
+import { UID2 } from "./uid2Sdk";
+import { isValidIdentity, Uid2Identity } from "./Uid2Identity";
+import { UID2CookieManager } from "./uid2CookieManager";
+import { UID2LocalStorageManager } from "./uid2LocalStorageManager";
+import { Uid2Options } from "./Uid2Options";
+
+export class UID2StorageManager {
+  private _cookieManager: UID2CookieManager | undefined;
+  private _localStorageManager: UID2LocalStorageManager | undefined;
+
+  private _opts: Uid2Options;
+  constructor(opts: Uid2Options) {
+    this._opts = opts;
+    this._cookieManager = new UID2CookieManager({ ...opts });
+    this._localStorageManager = new UID2LocalStorageManager();
+  }
+
+  public loadIdentityFromPreferredStorageWithFallback(): Uid2Identity | null {
```

Comment:
if we have `this._opts.useCookie` set to false, and we have cookieIdentity stored, do we still want to load identity from cookies?

Reply:
good question - actually this method name is misleading because it does not actually take `this._opts.useCookie` into consideration. I have renamed it to `loadIdentityWithFallback` to clarify this.

To answer your question, yes, I believe we still want to load the identity from the cookie. But then if `this._opts.useCookie` is false then it will, later on, save the value from the cookie into local storage and remove the cookie. 

Does that make sense?

Reply:
Oh, I see. This function is trying to find a fresh valid token no matter which storage it is stored in. I see the reason for that, thanks for the explanation.